tests/unit/hooks/state-machine.js

```diff
@@ -0,0 +1,269 @@
+'use strict';
+
+const assert = require('assert').strict;
+const camelCase = require('lodash.camelcase');
+const startCase = require('lodash.startcase');
+
+const { stateMachine } = require('../../..');
+
+const pascalCase = name => startCase(camelCase(name)).replace(/ /g, '');
+
+describe('Hooks', () => {
+
+	const serviceBase = {
+		custom: {},
+		plugins: [],
+		stepFunctions: {}
+	};
+
+	const definition = {
+		Comment: 'Create Session Machine',
+		StartAt: 'WaitForCall',
+		States: {
+			WaitForCall: {
+				Type: 'Wait',
+				SecondsPath: '$.body.wait',
+				Next: 'TrackingCall'
+			}
+		}
+	};
+
+	describe('State Machine', () => {
+
+		context('Config validation', () => {
+
+			it('Should throw if name param is missing', () => {
+
+				assert.throws(() => stateMachine(serviceBase, {
+					definition
+				}), {
+					message: /name/
+				});
+			});
+
+			it('Should throw if definition param is missing', () => {
+
+				assert.throws(() => stateMachine(serviceBase, {
+					name: 'my service'
+				}), {
+					message: /definition/
+				});
+			});
+		});
+
+		context('State Machine generation', () => {
+
+			it('Should return the service config without plugins, stepFunctions and the machines when passing the required params', () => {
+
+				const hooksParams = {
+					name: 'MachineName',
+					definition
+				};
+				const serviceConfig = stateMachine({}, hooksParams);
+
+				assert.deepStrictEqual(serviceConfig, {
+					plugins: [
+						'serverless-step-functions'
+					],
+					stepFunctions: {
+						stateMachines: {
+							[`SM-${pascalCase(hooksParams.name)}`]: hooksParams
+						}
+					},
+					custom: {
+						machines: {
+							MachineName: {
+								name: `SM-\${self:custom.serviceName}-${camelCase(hooksParams.name)}-\${self:custom.stage}`,
+								arn: {
+									'Fn::Join': [
+										':',
+										[
+											'arn:aws:states',
+											'${self:custom.region}',
+											{ Ref: 'AWS::AccountId' },
+											'stateMachine',
+											`\${self:custom.machines.${camelCase(hooksParams.name)}.name}`
+										]
+									]
+								}
+							}
+						}
+					}
+				});
+			});
+
+			it('Should return the service config and the state machine with name in camelCase', () => {
+
+				const hooksParams = {
+					name: 'Machine Name',
+					definition
+				};
+				const serviceConfig = stateMachine({}, hooksParams);
+
+				assert.deepStrictEqual(serviceConfig, {
+					plugins: [
+						'serverless-step-functions'
+					],
+					stepFunctions: {
+						stateMachines: {
+							[`SM-${pascalCase(hooksParams.name)}`]: hooksParams
```

Comment:
Te diría de reemplazar todos los tests que están así por el string hardcodeado para asegurarte que funciona todo bien..

```suggestion
							'SM-MachineName': hooksParams
```